src/hotspot/share/gc/g1/g1CollectedHeap.cpp

```diff
@@ -3101,7 +3104,7 @@ G1HeapRegion* G1CollectedHeap::new_mutator_alloc_region(size_t word_size,
   if (should_allocate) {
     G1HeapRegion* new_alloc_region = new_region(word_size,
                                                 G1HeapRegionType::Eden,
-                                                false /* do_expand */,
+                                                policy()->should_expand_on_mutator_allocation() /* do_expand */,
```

Comment:
On line 3102 above, there is an assert looking like this:
```assert_heap_locked_or_at_safepoint(true /* should_be_vm_thread */);```
The assumption seems to be that this is the VM thread. That won't be the case when exapnding from a mutator, right?